File: API/Core/Products/ProductService.cs
namespace Core.Products
{
    public interface IProductService
    {
        IReadOnlyCollection<Product> GetProducts();
    }

    public class ProductService : IProductService
    {
        private readonly IProductRepository productRepository;

        public ProductService(IProductRepository productRepository)
        {
            this.productRepository = productRepository;
        }

        public IReadOnlyCollection<Product> GetProducts() => productRepository.GetAll();
    }
}

File: client/src/app/products/product.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment'
import { HttpClient } from '@angular/common/http'
import { Observable } from 'rxjs';
import { product } from './Product';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private httpClient: HttpClient) { }

  getProducts() : Observable<product[]> {
    return this.httpClient.get<product[]>(`${environment.apiUrl}/products`);
  }
  
}
